scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,69 @@
+package com.twitter.scalding
+
+sealed abstract class DescribedArg {
+  def key: String
+  def description: String
+}
+
+case class RequiredArg(key: String, description: String) extends DescribedArg
+case class OptionalArg(key: String, description: String) extends DescribedArg
+case class ListArg(key: String, description: String) extends DescribedArg
+case class BooleanArg(key: String, description: String) extends DescribedArg
+
+object ArgHelp {
+  /**
+    * Describe the Arguments of this Execution.  By running --help the args will output
+    * and the execution will end
+    * @param describedArgs List of Argument Descriptions
+    * @param ex Input Execution
+    * @return Output Execution
+    */
+  def describe(describedArgs: Seq[DescribedArg], ex: Execution[Unit]): Execution[Unit] = {
+    Execution.getArgs.flatMap { args =>
+      if (args.boolean("help")) {
+        val top = "\n###########################################################################\n\n"
+        val usage = s"Command Line Args :: ${argString(describedArgs)}\n\n\n"
+        val help = help(describedArgs)
+        val bottom = "\n\n###########################################################################\n"
+        println(top + usage + help + bottom)
+        Execution.unit
+      } else {
+        ex
+      }
+    }
+  }
+
+  /**
+    * Command line arg string given the Described Args
+    * @param describedArgs List of Argument Descriptions
+    * @return Command Line Parameters
+    */
+  def argString(describedArgs: Seq[DescribedArg]): String = {
```

Comment:
This isn't used now, is it needed with the help string? is it maybe the usage string?